executor/src/witgen/global_constraints.rs

```diff
@@ -338,35 +347,25 @@ namespace Global(2**20);
 ";
         let analyzed = pil_analyzer::analyze_string::<GoldilocksField>(pil_source);
         let (constants, _) = crate::constant_evaluator::generate(&analyzed);
-        let fixed_polys = constants
-            .iter()
-            .enumerate()
-            .map(|(i, (name, _))| PolynomialReference {
-                name: name.to_string(),
-                poly_id: Some(PolyID {
-                    id: i as u64,
-                    ptype: PolynomialType::Constant,
-                }),
-                index: None,
-                next: false,
-            })
+        let fixed_polys = (0..constants.len())
+            .map(|i| constant_poly_id(i as u64))
             .collect::<Vec<_>>();
         let mut known_constraints = fixed_polys
             .iter()
             .zip(&constants)
-            .filter_map(|(poly, (_, values))| {
-                process_fixed_column(values).map(|(constraint, _full)| (poly, constraint))
+            .filter_map(|(&poly_id, (_, values))| {
+                process_fixed_column(values).map(|(constraint, _full)| (poly_id, constraint))
             })
             .collect::<BTreeMap<_, _>>();
         assert_eq!(
-            known_constraints
-                .iter()
-                .map(convert_constraints)
-                .collect::<BTreeMap<_, _>>(),
+            known_constraints,
             vec![
-                ("Global.BYTE", RangeConstraint::from_max_bit(7)),
-                ("Global.BYTE2", RangeConstraint::from_max_bit(15)),
-                ("Global.SHIFTED", RangeConstraint::from_mask(0xff0_u32)),
+                // Global.BYTE
+                (constant_poly_id(0), RangeConstraint::from_max_bit(7)),
```

Comment:
So this is because `PolyID` cannot format itself, and adding as reference to the name is not possible because of lifetimes?

Reply:
Yeah, basically. I wouldn't say it's impossible to add the reference, just a bigger change.